docs/API.md

```diff
@@ -263,6 +263,56 @@ Request body:
 }
 ```
 
+### Charm and bundle publishing
+
+#### PUT *id*/publish
+
+A PUT to ~*user*/*anyseries*/*name*-*anyrevision* sets whether the
+corresponding charm or bundle is published and can be accessed through a URL
+with no channel. If the revision number is not specified, the id is resolved to
+the charm or bundle with the latest development revision number. The id must
```

Comment:
Looking at the code, this doesn't appear to be true when unpublishing - in that case the latest published revision number is chosen. If we want that to be the case (seems kinda reasonable, I guess) then we should document that.